tools/gradle-plugin/src/test/kotlin/io/github/composegears/valkyrie/gradle/Utils.kt

```diff
@@ -0,0 +1,47 @@
+package io.github.composegears.valkyrie.gradle
+
+import java.io.File
+import org.gradle.testkit.runner.GradleRunner
+
+internal fun runTask(root: File, task: String) = GradleRunner
+    .create()
+    .withPluginClasspath()
+    .withGradleVersion(System.getProperty("test.version.gradle"))
+    .withProjectDir(root)
+    .withArguments(task, "--configuration-cache", "--info", "--stacktrace")
+
+internal fun File.writeSettingsFile() = resolve("settings.gradle.kts").writeText(
+    """
+        pluginManagement {
+            repositories {
+                mavenCentral()
+                google()
+                gradlePluginPortal()
+            }
+        }
+
+        dependencyResolutionManagement {
+            repositories {
+                mavenCentral()
+                google()
+            }
+        }
+    """.trimIndent(),
+)
+
+internal fun File.writeLibsTomlFile() {
+    val agp = System.getProperty("test.version.agp")
+    val kotlin = System.getProperty("test.version.kotlin")
```

Comment:
There is a more convenient and useful way to achieve it, try out things like https://github.com/GradleUp/shadow/blob/1d7b0863fed3126bf376f11d563e9176de176cd3/build.gradle.kts#L204-L209, and you'll get rid of the `ibs.versions.toml`.

Reply:
Cool - just updated it so the tests pass with gradle-api. `agp-full` is being used in testPluginClasspath - I assume that's okay for now?

Reply:
Well done!